extensions/panache/panache-mock/deployment/src/main/java/io/quarkus/panache/mock/deployment/PanacheMockMethodCustomizer.java

```diff
@@ -0,0 +1,123 @@
+         * } catch (PanacheMock.InvokeRealMethodException e) {
+         * // fall-through
+         * }
+         * }
+         * 
+         * Bytecode approx:
+         * 
+         * 0: getstatic #16 // Field PanacheMock.IsMockEnabled:Z
+         * 3: ifeq 50
+         * 6: ldc #1 // class MyTestMockito$TestClass
+         * 8: invokestatic #22 // Method PanacheMock.isMocked:(Ljava/lang/Class;)Z
+         * 11: ifeq 50
+         * 14: ldc #1 // class MyTestMockito$TestClass
+         * 16: ldc #26 // String foo
+         * 
+         * 18: iconst_1
+         * 19: anewarray #27 // class java/lang/Class
+         * 22: dup
+         * 23: iconst_0
+         * 24: getstatic #29 // Field java/lang/Integer.TYPE:Ljava/lang/Class;
+         * 27: aastore
+         *
+         * 28: iconst_1
+         * 29: anewarray #3 // class java/lang/Object
+         * 32: dup
+         * 33: iconst_0
+         * 34: iload_0
+         * 35: invokestatic #35 // Method java/lang/Integer.valueOf:(I)Ljava/lang/Integer;
+         * 38: aastore
+         * 
+         * 39: invokestatic #39 // Method
+         * PanacheMock.mockMethod:(Ljava/lang/Class;Ljava/lang/String;[Ljava/lang/Class;[Ljava/lang/Object;)Ljava/lang/Object;
+         * 42: checkcast #30 // class java/lang/Integer
+         * 45: invokevirtual #43 // Method java/lang/Integer.intValue:()I
+         * 48: ireturn
+         * 49: astore_1
+         */
+        Label realMethodLabel = new Label();
+
+        mv.visitFieldInsn(Opcodes.GETSTATIC, PANACHE_MOCK_BINARY_NAME, "IsMockEnabled", "Z");
+        mv.visitJumpInsn(Opcodes.IFEQ, realMethodLabel);
+
+        mv.visitLdcInsn(entityClassSignature);
+        mv.visitMethodInsn(Opcodes.INVOKESTATIC, PANACHE_MOCK_BINARY_NAME, "isMocked", "(Ljava/lang/Class;)Z", false);
+        mv.visitJumpInsn(Opcodes.IFEQ, realMethodLabel);
+
+        Label tryStart = new Label();
+        Label tryEnd = new Label();
+        Label tryHandler = new Label();
+        mv.visitTryCatchBlock(tryStart, tryEnd, tryHandler, PANACHE_MOCK_INVOKE_REAL_METHOD_EXCEPTION_BINARY_NAME);
+        mv.visitLabel(tryStart);
+
+        mv.visitLdcInsn(entityClassSignature);
+        mv.visitLdcInsn(method.name());
+
+        mv.visitLdcInsn(method.parameters().size());
+        mv.visitTypeInsn(Opcodes.ANEWARRAY, "java/lang/Class");
+
+        int i = 0;
+        for (org.jboss.jandex.Type paramType : method.parameters()) {
+            mv.visitInsn(Opcodes.DUP);
+            mv.visitLdcInsn(i);
+            JandexUtil.visitLdc(mv, paramType);
+            mv.visitInsn(Opcodes.AASTORE);
+            i++;
+        }
+
+        mv.visitLdcInsn(method.parameters().size());
+        mv.visitTypeInsn(Opcodes.ANEWARRAY, "java/lang/Object");
+
+        i = 0;
+        for (org.jboss.jandex.Type paramType : method.parameters()) {
+            mv.visitInsn(Opcodes.DUP);
+            mv.visitLdcInsn(i);
```

Comment:
Double and long take up two local variable slots. Have you tested this methods methods that take double and long as params? Whenever you are doing anything with bytecode that involves primitives it is always a good idea to write explicit checks for long and double as they are a pain.

Reply:
So, we don't have any method in the superclasses in question that we generate bridges for (`PanacheEntityBase` and `PanacheRepositoryBase`) so it doesn't happen. And I can't test this by adding a method to these classes, and no other classes have bridge methods generated.

So in general, sure, but I couldn't test this if I supported it. I could add a guard to make sure we never add such methods.

Reply:
You were right, I've implemented and tested it.